src/NuGet.Services.BasicSearch/ApplicationInsightsMiddleware.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Owin;
+
+namespace NuGet.Services.BasicSearch
+{
+    public class ApplicationInsightsMiddleware
+        : OwinMiddleware
+    {
+        public const string OwinRequestIdKey = "owin.RequestId";
+
+        private readonly TelemetryClient _telemetryClient;
+
+        public ApplicationInsightsMiddleware(OwinMiddleware next)
+            : this(next, null)
+        {
+        }
+
+        public ApplicationInsightsMiddleware(OwinMiddleware next, TelemetryConfiguration telemetryConfiguration) 
+            : base(next)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
```

Comment:
Removed the check. Ideally this should be (one of) the first middlewares registered. Good catch!